spec/services/source_refresh_service_spec.rb

```diff
@@ -0,0 +1,75 @@
+describe SourceRefreshService do
+  include ::Spec::Support::TenantIdentity
+  let(:subject) { described_class.new(source) }
+
+  let!(:refresh_task) { FactoryBot.create(:task, :tenant => tenant, :created_at => Time.current, :state => state, :child_task_id => child_task_id) }
```

Comment:
@hsong-rh What happens if the child_task_id points to a non existing object, which has been deleted by cronjob.